### DISCUSSION TITLE
React Native Web3Modal Not Opening

### DISCUSSION THREAD

**Discussion by xgambitox** [WalletConnect/walletconnect-monorepo#2348] [Developer Support] [answered]
Hey all! Im following the current efforts to bring the Web3Modal to react native through https://github.com/WalletConnect/web3modal-react-native. 

I created a new project to test it out, following these instructions:
https://docs.walletconnect.com/2.0/reactnative/web3modal/installation
https://docs.walletconnect.com/2.0/reactnative/web3modal/dapp-usage

My issue is the `Web3Button` seems to always be in a loading state. Also using `open` function from `useWeb3Modal` hook doesnt trigger anything for me.

Im using expo v48.0.15 through WSL2. These are the dependencies from a fresh install:
```
"dependencies": {
    "@react-native-async-storage/async-storage": "1.17.11",
    "@web3modal/react-native": "^1.0.0-alpha.3",
    "expo": "~48.0.15",
    "expo-crypto": "~12.2.1",
    "expo-status-bar": "~1.4.4",
    "react": "18.2.0",
    "react-native": "0.71.7",
    "react-native-get-random-values": "~1.8.0",
    "react-native-modal": "^13.0.1",
    "react-native-svg": "13.4.0"
  },
```
any idea?

**Comment by finessevanes** [comment]
Hey - thank you for posting. Can you share your code implementation?

**Reply by xgambitox** [reply, marked as answer]
Hey @finessevanes, thanks for responding, I was able to find what was causing the issue. The `Web3Modal` component needs a `providerMetadata` prop as following:
```js
<Web3Modal projectId={projectId} providerMetadata={providerMetadata} />
```
Might be worth adding to the docs here https://docs.walletconnect.com/2.0/reactnative/web3modal/dapp-usage#implementation

**Reply by finessevanes** [reply]
TY! https://github.com/WalletConnect/walletconnect-docs/pull/678 🫡
